### ISSUE TITLE
GCHandle.IsAllocated documentation appears to be incorrect

### ISSUE THREAD

**Issue by PathogenDavid** [dotnet/dotnet-api-docs#4909]
The [`GCHandle.IsAllocated` remarks](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.gchandle.isallocated?view=netcore-3.1#remarks) states the following:

> Use this property when using Weak handles to determine if the GCHandle is still available. When the garbage collector collects the object, the Weak handle can still be resurrected in the finalizer. In that case, the handle is not allocated (it is lost when the garbage collector attempts to collect the object), even though the target object is valid.

The [implementation of `IsAllocated`](https://github.com/dotnet/runtime/blob/9c472545df220d7ae26329e1649f880c4ee127e8/src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/GCHandle.cs#L144) only checks whether its internal handle is null:

```csharp
public bool IsAllocated => _handle != IntPtr.Zero;
```

(The [.NET Framework implementation](https://referencesource.microsoft.com/#mscorlib/system/runtime/interopservices/gchandle.cs,0bda1329b5d1b7ac) is basically identical.)

Unless there's some magic handling of `GCHandle` in the runtime, I believe the only time `_handle` is null (and `IsAllocated` is false) is if the handle was never initialized or the actual handle was freed with `Free()`.

Weak handles to freed objects are still allocated, they just don't point to anything (`Target` returns null.)

[Empirical tests seem to confirm all of this.](https://gist.github.com/PathogenDavid/b541e1615631c50203770531f4cb0aa5)

Is this documentation incorrect or am I missing something?

**Comment by msedi** [comment]
I agree with @PathogenDavid. I'm found the problem because I couldn't use a `WeakReference<T>`. So it would be nice to hear if the docs are incorreect or the behavior is wrong.

**Comment by reflectronic** [comment]
@jkotas Can you clarify the expected behavior?

**Comment by jkotas** [comment]
Yes, the documentation is completely wrong.

IsAllocated is false is if the handle was never initialized or the handle was freed with Free.

Would you like to submit a PR with doc update?
